src/test/java/com/conveyal/gtfs/validator/FlexValidatorTest.java

```diff
@@ -100,192 +141,285 @@ private static Stream<LocationArguments> createLocationChecks() {
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_LOCATION_GROUP_ID)
             ),
             new BaseArguments(
-                createStopTime(INT_MISSING, 1330, 1100, INT_MISSING),
-                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_DEPARTURE_TIME, NewGTFSErrorType.FLEX_FORBIDDEN_START_PICKUP_DROP_OFF_WINDOW)
+                createStopTimeForStopIdTest("stop-id-1", "location-group-id-1", null),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_LOCATION_GROUP_ID)
             ),
             new BaseArguments(
-                createStopTime(INT_MISSING, 1330, INT_MISSING, 1200),
-                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_DEPARTURE_TIME, NewGTFSErrorType.FLEX_FORBIDDEN_END_PICKUP_DROP_OFF_WINDOW)
+                createStopTimeForStopIdTest(null, "location-group-id-1", "location-id-1"),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_LOCATION_GROUP_ID)
             )
         );
     }
 
     @ParameterizedTest
-    @MethodSource("createStopTimeChecksForStartEndPickupDropoff")
-    void validateStopTimeStartEndPickupDropoffTests(StopTimeArguments stopTimeArguments
-    ) {
-        List<NewGTFSError> errors = FlexValidator.validateStopTime(
-            (StopTime) stopTimeArguments.testObject,
-            stopTimeArguments.stopAreas,
-            stopTimeArguments.locations
+    @MethodSource("createLocationIdTests")
+    void validateLocationId(BaseArguments baseArguments) {
+        List<NewGTFSError> errors = new ArrayList<>();
+        FlexValidator.validateLocationId((StopTime) baseArguments.testObject, errors);
+        checkValidationErrorsMatchExpectedErrors(errors, baseArguments.expectedErrors);
+    }
+
+    private static Stream<BaseArguments> createLocationIdTests() {
+        return Stream.of(
+            new BaseArguments(
+                createStopTimeForStopIdTest(null, null, "location-id-1"), null
+            ),
+            new BaseArguments(
+                createStopTimeForStopIdTest(null, null, null),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_LOCATION_ID)
+            ),
+            new BaseArguments(
+                createStopTimeForStopIdTest("stop-id-1", null, "location-id-1"),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_LOCATION_ID)
+            ),
+            new BaseArguments(
+                createStopTimeForStopIdTest(null, "location-group-id-1", "location-id-1"),
+                Lists.newArrayList(NewGTFSErrorType.FLEX_FORBIDDEN_LOCATION_ID)
+            )
         );
-        checkValidationErrorsMatchExpectedErrors(errors, stopTimeArguments.expectedErrors);
     }
 
-    private static Stream<StopTimeArguments> createStopTimeChecksForStartEndPickupDropoff() {
+    @ParameterizedTest
+    @MethodSource("createStartPickupDropOffWindowTests")
+    void validateStartPickupDropOffWindowTest(BaseArguments baseArguments) {
```

Comment:
You could spread `baseArguments` and put its attributes as this method's arguments instead (many other instances)